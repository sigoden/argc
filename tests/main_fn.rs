use crate::*;

#[test]
fn with_main() {
    let script = r###"
# @arg val
main() { :? }
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn no_main() {
    let script = r###"
# @arg val
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn subcmd_main() {
    let script = r###"
# @cmd
cmd() { :? }
main() { :? }
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn subcmd_no_main() {
    let script = r###"
# @cmd
cmd() { :? }
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn nested_subcmd_main() {
    let script = r###"
# @cmd
cmd() { :? }
cmd::main() { :? }
# @cmd
cmd::foo() { :? }
"###;
    snapshot!(script, &["prog", "cmd"]);
}

#[test]
fn nested_subcmd_no_main() {
    let script = r###"
# @cmd
cmd() { :? }
# @cmd
cmd::foo() { :? }
"###;
    snapshot!(script, &["prog", "cmd"]);
}
