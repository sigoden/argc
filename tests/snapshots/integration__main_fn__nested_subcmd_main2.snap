---
source: tests/main_fn.rs
expression: data
---
************ RUN ************
prog cmd

# OUTPUT
argc__args=( prog cmd )
argc__fn=cmd::main
argc__positionals=(  )
cmd::main

# RUN_OUTPUT
argc__args=([0]="prog" [1]="cmd")
argc__fn=cmd::main
argc__positionals=()
cmd::main

************ RUN ************
prog cmd abc --foo 123

# OUTPUT
argc_foo=123
argc_val=abc
argc__args=( prog cmd abc --foo 123 )
argc__fn=cmd::main
argc__positionals=( abc )
cmd::main abc

# RUN_OUTPUT
argc__args=([0]="prog" [1]="cmd" [2]="abc" [3]="--foo" [4]="123")
argc__fn=cmd::main
argc__positionals=([0]="abc")
argc_foo=123
argc_val=abc
cmd::main abc


